src/boot/efi/sha256.h

```diff
@@ -1,6 +1,8 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#include <endian.h>
```

Comment:
that's a glibc header, no? this is EFI code, we cannot just include glibc headers here. what am i missing?

Reply:
actually, this looks wrong entirely. __BYTE_ORDER__ and __ORDER_LITTLE_ENDIAN__ are defined by gcc itself, not glibc. i.e.

https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html#Common-Predefined-Macros

endian defines `__BYTE_ORDER`, i.e. the subtle __ suffix is not there...

Reply:
#19802.